### ISSUE TITLE
RGroupDecomp can be a bit wordy

### ISSUE THREAD

**Issue by bp-kelley** [rdkit/rdkit#3962]
@ptosco, @jones-gareth 

When decomposing a molecule, the r-group decomp spits out warnings like to the log like:

"No core match"
"No attachment points in side chains"

and returns -1 in both cases.  I would rather make these debug notifications and return proper error codes.

I.e. 

```
  if (potentialMatches.size() == 0) {
    BOOST_LOG(rdDebugLog)
        << "No attachment points in side chains" << std::endl;
    return RGroupErrors.NoAttachmentPoints;
  }
```

```
  if (rcore == nullptr) {
   BOOST_LOG(rdDebugLog) << "No core matches" << std::endl;
    return RGroupErrors.NoCoreMatch;
  }
```

At the minimum, I would like to get rid of the errors as I don't really think this is an error state, especially when running on larger decompositions.

**Comment by ptosco** [comment]
I think having more granular error codes is a very good idea.

**Comment by jones-gareth** [comment]
Good idea.

Because negative score are used by  fingerprint variance scoring, -1 doesn't work as an error indicator.  The process functions now return a `RGroupDecompositionProcessResult` object that has a score field and a success field.

**Comment by bp-kelley** [comment]
These return values are for adding a molecule, so they shouldn’t affect scoring.

**Comment by jones-gareth** [comment]
> 
> 
> These return values are for adding a molecule, so they shouldn’t affect scoring.

Of course they are!  Sorry about that.

**Comment by greglandrum** [comment]
I'm fine with making the code more verbose and/or changing the values returned
